File: frontend/src/types/plane/plane.dto.ts

export interface PlaneDto {
    callSign: string,
    aircraft: AircraftDto,
    status: string
}

export interface AircraftDto {
    manufacturer: string,
    model: string,
    cruiseSpeedKnots: number,
    rangeNauticalMiles: number,
    passengerCapacity: number,
    category: string
}

File: Airline.Server/src/Api/Dtos/PlaneDto.cs
using Airline.Server.Domain.fleet.Entity;

namespace Airline.Server.Api.Dtos;

public class PlaneDto
{
    public string CallSign { get; set; }
    public AircraftDto Aircraft { get; set; }
    public string Status { get; set; }

    public PlaneDto(Plane plane)
    {
        CallSign = plane.CallSign;
        Aircraft = new AircraftDto(plane.Aircraft);
        Status = plane.Status.ToString();
    }
}